src/actions.ts

```diff
@@ -581,6 +585,129 @@ function meActions(
+
+				{
+					const inNextSelection =
+						currentNextSelection.length > 0 && currentNextSelection[0] === Enums.TransitionSelection.Background
+					if (inNextSelection) i++
+
+					let include
+					const mode = action.options.background as NextTransBackgroundChoices
+					switch (mode) {
+						case NextTransBackgroundChoices.NoChange:
+							include = inNextSelection
+							break
+						case NextTransBackgroundChoices.Include:
+							include = true
+							break
+						case NextTransBackgroundChoices.Omit:
+							include = false
+							break
+						case NextTransBackgroundChoices.Toggle:
+							include = !inNextSelection
+							break
+						default:
+							assertUnreachable(mode)
+							instance.log('debug', 'Unknown background mode: ' + mode)
+							include = inNextSelection
+							break
+					}
+
+					if (include) nextSelection.push(Enums.TransitionSelection.Background)
+				}
+
+				for (let key = 0; key < model.USKs; key++) {
+					const usk = getUSK(state.state, mixeffect, key)
+					if (usk) {
+						const component = 1 << (key + 1)
+						const inNextSelection = i < currentNextSelection.length && currentNextSelection[i] === component
+						if (inNextSelection) i++
+
+						const mode = action.options[`key${key}`] as NextTransKeyChoices
+						let include
+						switch (mode) {
+							case NextTransKeyChoices.NoChange:
+								include = inNextSelection
+								break
+							case NextTransKeyChoices.On:
+							case NextTransKeyChoices.Off:
+								include = usk.onAir === (mode === NextTransKeyChoices.Off)
+								break
+							case NextTransKeyChoices.Toggle:
+								include = !inNextSelection
+								break
+							case NextTransKeyChoices.Include:
+								include = true
+								break
+							case NextTransKeyChoices.Omit:
+								include = false
+								break
+							default:
+								assertUnreachable(mode)
+								instance.log('debug', 'Unknown mode: ' + mode)
+								include = inNextSelection
+								break
+						}
+
+						if (include) nextSelection.push(component)
+					}
+				}
+
+				if (nextSelection.length > 0) {
+					// If no components are selected, do nothing.
+					await atem.setTransitionStyle({ nextSelection }, mixeffect)
+				}
+			},
+			learn: (action) => {
+				const me = action.options.mixeffect
+				const mixeffect = getMixEffect(state.state, me)
+				const tp = getTransitionProperties(state.state, me)
+				if (mixeffect && tp) {
+					const currentNextSelection = tp?.nextSelection.sort((a, b) => a - b)
+					let i = 0
+
+					let background
+					if (i < currentNextSelection.length && currentNextSelection[i] === Enums.TransitionSelection.Background) {
+						i++
+						background = NextTransBackgroundChoices.Include
+					} else {
+						background = NextTransBackgroundChoices.Omit
+					}
+
+					const keys: { [key: string]: InputValue } = {}
+					for (let key = 0; key < model.USKs; key++) {
+						const usk = getUSK(state.state, me, key)
+						if (!usk) return undefined
```

Comment:
Should this be returning if usk is undefined?
I think it would be better to use some default value instead (`Omit` maybe?), so that the rest of the properties can be learnt as expected.

Reply:
Re the first question:

I'm having a doozy of a time determining from Github search-by-references exactly how `state.state.video.mixEffects[key]` is initialized.  It doesn't appear this package does it.  Maybe only https://github.com/nrkno/sofie-atem-connection/blob/9f87cc1464e92e32adf248c3385dcbf964caa260/src/state/util.ts#L117 does?  (It appears that every caller calls that function without the trailing argument, so the mix effect is always created if looked up.)

My assumption is _somewhere_ in this mild morass is a guarantee that `key < model.USKs` implies the atem-connection's `getMixEffect` has been called.  I think this is correct because every call of that function appears in an `applyToState` that's preceded by an in-range check.  And this code runs after an application-to-state.

So I think it's impossible to have `usk === undefined` here.

Re the second comment:

If the conclusion on the first question is correct, this is needless defensiveness that doesn't matter.  But if that conclusion isn't correct, sure, `Omit` seems fine.  I'll do `Omit`.

Reply:
True, it probably is impossible for `usk === undefined` to ever be true here, so maybe a comment saying that would have been sufficient. But that wont necessarily remain impossible forever, and this overly defensive is preventing future bugs. Its an easy and cheap change to make, so I see no harm in doing it while we remember the intention.